### PR TITLE
Feature/argodev 423

### PR DESCRIPTION
**Pull request by qtl-bodc**
updated processo2.m to include calls to mdoxy2doxy for the switch case 201_201_301 and 202_201_301

### CODE REVIEW

**Review by vidkris — CHANGES_REQUESTED** [review]
Looking good. Just a small doubts which I had.

**Review by vidkris — COMMENTED** [review]
looking really good. some small doubts that i had.

File: bgc_derivation/processo2.m [hunk 1]
```diff
@@ -117,28 +121,62 @@ function processo2(floatnos,varargin)
...
+          P  =  ncctd{'PRES'}(1,:);
+          S  =  ncctd{'PSAL'}(1,:);
+          molar_doxy = ncdox{'MOLAR_DOXY'}(1,:);
+          doxy = ncdox{'DOXY'}(1,:);
 
-          doxycalc  =  molar_doxy;             % Get right dimensions
-          doxycalc(mask)  =  O2ctoO2s(molar_doxy(mask),T(mask),S(mask),P(mask)); %,p_atm(mask));
-          doxy  =  ncdox{'DOXY'}(:)';
+          fillval_doxy = ncreadatt(pathdox{ii},'MOLAR_DOXY','_FillValue');
+          fillval_ctd = ncreadatt(pathctd{ii},'PSAL','_FillValue');
```
[the remark this answers is not in the document]
Comment (qtl-bodc) [reply bgc_derivation/processo2.m]: Those lines have been commented out in the latest commit. But the _FillValue represents the 'Absent' data values i.e. the 99999 original set by Steve to create the mask variable which isn't used anymore.

File: bgc_derivation/processo2.m [hunk 2]
```diff
@@ -117,28 +121,62 @@ function processo2(floatnos,varargin)
...
+          %
+          if isempty(stcoeff)
+            %{ % Argo manual defaults
+            coeffs.Sref=0; coeffs.Spreset=0;
+            coeffs.Pcoef2=0.00025; coeffs.Pcoef3=0.0328;
+            coeffs.B0=-6.24523e-3; coeffs.B1=-7.37614e-3; coeffs.B2=-1.03410e-2; coeffs.B3=-8.17083e-3;
+            coeffs.C0=-4.88682e-7;
+            coeffs.D0=24.4543; coeffs.D1=-67.4509; coeffs.D2=-4.8489; coeffs.D3=-0.000544;
+            %}
+            disp('could not find predeployment calibration coefficients')
```
[the remark this answers is not in the document]
Comment (qtl-bodc) [reply bgc_derivation/processo2.m:140]: The defaults are not set. they are actually commented out. Violetta said (See Argo chat channel) that any floats without calibration information should not be used. So the code acutally skips those floats.

File: bgc_derivation/processo2.m [hunk 3]
```diff
@@ -117,28 +121,62 @@ function processo2(floatnos,varargin)
...
-          P  =  ncctd{'PRES_ADJUSTED'}(:)';
-          S  =  ncctd{'PSAL_ADJUSTED'}(:)';
+        case {'201_201_301','202_201_301'}
 
-          %  NB THIS CODE is under development - stored and calculated values
-                              %  do not yet agree
-          molar_doxy  =  ncdox{'MOLAR_DOXY'}(:)';
-          maskdoxy  =  molar_doxy  ==  99999;  %  More properly this should be the fill value
-          maskctd  =  S == 99999;              %  as defined in the .nc file
+          % if empty coefficients then either we can
```
[the remark this answers is not in the document]
Comment (qtl-bodc) [reply bgc_derivation/processo2.m:126]: If there are not co-efficients in the netcdf file, then the float is skipped. Violetta said it wasn't safe to assume the floats were set to use factory defaults. (See ARGO chat channel)

**Pull request merged**